packages/odie-client/src/index.tsx

```diff
@@ -12,16 +13,34 @@ export const ODIE_THUMBS_DOWN_RATING_VALUE = 0;
 export const ODIE_THUMBS_UP_RATING_VALUE = 1;
 
 export const OdieAssistant: React.FC = () => {
-	const { chat, trackEvent, currentUser, isLoadingEnvironment, isLoadingExistingChat } =
-		useOdieAssistantContext();
+	const {
+		chat,
+		trackEvent,
+		currentUser,
+		isLoadingEnvironment,
+		isLoadingExistingChat,
+		isUserElegible,
+	} = useOdieAssistantContext();
 	const [ asisstantLoaded, setAssistantLoaded ] = useState( false );
 	const containerRef = useRef< HTMLDivElement >( null );
 	const messagesContainerRef = useRef< HTMLDivElement >( null );
+	const navigate = useNavigate();
 
 	useEffect( () => {
 		trackEvent( 'chatbox_view' );
 	}, [ trackEvent ] );
 
+	// Prevent not eligible users from accessing wapuu.
+	useEffect( () => {
+		if ( ! isUserElegible ) {
```

Comment:
I'm worried we may redirect a user who is still loading support status. We should probably check that it's not loading too.

Reply:
The way we are injecting the isUserElegible is through the query:

```tsx
	const { data } = useSupportStatus();
	const isUserElegible = data?.eligibility.is_user_eligible ?? false;
```

So I would say that particular case is covered, but we could add anyway something like:
```tsx
	const { data, isLoading } = useSupportStatus();
	const isUserElegible = isLoading ? : false : data?.eligibility.is_user_eligible ?? false;
```

But I don't think it's necessary

Reply:
Is it covered? The user will seem illegible until the query loads, no?